### PR TITLE
[FLINK-22393][docs-zh] Translate the page of "Execution Mode (Batch/Streaming)" into Chinese

### PR DESCRIPTION
**Pull request by 95chenjz**
## What is the purpose of the change

Translate the page of "Execution Mode (Batch/Streaming)" into Chinese

The page url of "Execution Mode (Batch/Streaming)" is https://ci.apache.org/projects/flink/flink-docs-release-1.12/zh/dev/datastream_execution_mode.html.

The markdown file can be found in flink/docs/content.zh/docs/dev/datastream/execution_mode.md in English.

## Brief change log

- Translate 'flink/docs/content.zh/docs/dev/datastream/execution_mode.md'.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no 
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not documented

### CODE REVIEW

**Comment by 95chenjz** [comment]
cc @wuchong @xccui @klion26  @PatrickRen @becketqin. Could you review my PR when you're free? No hurry, looking for your  feedback.

**Comment by PatrickRen** [comment]
Thanks for the contribution @95chenjz ! But it looks like a new `.md` file is created for the Chinese documentation. I think the translation should be made in the file `docs/content.zh/docs/dev/datastream/execution_mode.md` instead of creating a new one. 

Also the `README.md` under `docs` describes how to build the doc and view locally on your browser. You can have a preview to see if your modifications take effect.

**Comment by 95chenjz** [comment]
> Thanks for the contribution @95chenjz ! But it looks like a new `.md` file is created for the Chinese documentation. I think the translation should be made in the file `docs/content.zh/docs/dev/datastream/execution_mode.md` instead of creating a new one.
> 
> Also the `README.md` under `docs` describes how to build the doc and view locally on your browser. You can have a preview to see if your modifications take effect.

Thank you for your comment. It's fixed.

**Comment by 95chenjz** [comment]
@PatrickRen Hi Patrick, is everything looking alright now?

File: docs/content.zh/docs/dev/datastream/execution_mode.md [hunk 1]
```diff
@@ -161,235 +106,125 @@ source.name("source")
...
+你仍然可以使用所有的 [状态原语（state primitives）]({{< ref "docs/dev/datastream/fault-tolerance/state" >}}#working-with-state)，只是用于故障恢复的机制会有所不同。
 
-### Writing Custom Operators
+### 编写自定义算子
 
 {{< hint info >}}
-**Note:** Custom operators are an advanced usage pattern of Apache Flink. For most
-use-cases, consider using a (keyed-)process function instead.
-{{< /hint >}}
+**注意：** 自定义算子是 Apache Flink 的一种高级使用模式。对于大多数的使用情况，可以考虑使用（keyed-）过程函数来代替。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/execution_mode.md:225]: “可以考虑使用（keyed-）~~过程~~ 处理 函数来代替”

**Review by PatrickRen — COMMENTED** [review]
Hi @95chenjz , this translation looks good to me, only some typos and trivial problems. I left some comments. 

Also I don't have the permission to merge PRs, you can ping @wuchong after these typos are fixed. 

Thanks again for the contribution!

File: docs/content.zh/docs/dev/datastream/execution_mode.md (hunk 1 above)
Comment (95chenjz) [reply docs/content.zh/docs/dev/datastream/execution_mode.md]: Thanks again for your review and comments! I have my PR updated. @wuchong  Could you help to merge?

**Comment by 95chenjz** [comment]
> Hi @95chenjz , this translation looks good to me, only some typos and trivial problems. I left some comments.
> 
> Also I don't have the permission to merge PRs, you can ping @wuchong after these typos are fixed.
> 
> Thanks again for the contribution!

Hi @PatrickRen thanks for your comments. I think it's all set. @wuchong is there anything else I should do?

**Comment by wuchong** [comment]
@PatrickRen once you are fine with the translation. I can help to merge this.

**Comment by PatrickRen** [comment]
@wuchong I think the translation is ready to be merged.

**Review by PatrickRen — APPROVED** [review]
@wuchong I think the translation is ready to be merged.

**Pull request merged**